src/main/java/amidst/util/SwingUtils.java

```diff
@@ -0,0 +1,91 @@
+package amidst.util;
+
+import java.awt.Component;
+import java.awt.Container;
+import java.lang.reflect.Method;
+import java.util.EventListener;
+
+import javax.swing.MenuElement;
+
+import amidst.documentation.AmidstThread;
+import amidst.documentation.CalledOnlyBy;
+
+public enum SwingUtils {
+	;
+
+	/**
+	 * This helps with allowing swing components to get garbage collected.
+	 */
+	@CalledOnlyBy(AmidstThread.EDT)
+	public static void destroyComponentTree(Component c) {
+		// calls the component's dispose method if it has it.
+		try {
+			Method m = c.getClass().getMethod("dispose");
+			m.invoke(c);
+		} catch (Exception e) {
+		}
```

Comment:
I think only `java.awt.Window` has a `dispose()` method, so this could probably be replaced by an `instanceof Window` check.

Reply:
Fixed. I could see the possibility of a custom component having a method called `dispose()`, but I guess it would be the job of that component to do it itself anyway.